### DISCUSSION TITLE
Config Tool

### DISCUSSION THREAD

**Discussion by NeoCHoZen** [danielnilsson9/bbs-fw#38] [General]
Hey Daniel, me again :D 

Quick question about the config tool. I noticed on my original stock configuration (pre flash), I had both an eco mode and a sport mode, each with 5 levels of assist and throttle. After the flash, (which was successful thank you!), I have 5 levels of eco mode and throttle only. I can switch to sport, but there's no assist or throttle. I'm assuming this has to be configured within the config tool right? Based on your answer, there may be more questions. Any help is appreciated!

Tim

**Comment by NeoCHoZen** [comment]
Hold this thought. I see where you explain it in the documentation. If I have any issues, I'll let you know. 

Thanks,
Tim

**Comment by CitizenDev** [comment]
> I noticed on my original stock configuration (pre flash), I had both an eco mode and a sport mode, each with 5 levels of assist and throttle.

How did you have 2 separate assist configuration modes on same original firmware? To my knowledge this is only available in this custom firmare and EggRider V2 display unit. What kind of display unit u used?

**Reply by NeoCHoZen** [reply]
My display is a DCP-18. Holding the up arrow on the bar control switches between eco mode and sport mode. Each mode has 5 levels of assist and the sport mode is noticeably more powerful and uses more battery than the eco mode. I say original firmware to mean the firmware that came with my BBSHD setup from the vendor.

**Reply by danielnilsson9** [reply]
Interesting, never thought that that mode change did anything on the BBSHD, at least it didn't do anything when I bought my first motor in 2017. But that may have changed in later firmware revisions.

As far as I know it was not possible to configure two sets of different assist levels as there is only one page in the configuration tool. So the mode switch must have meant a fixes power decrease factor was applied when switching to eco or something.

Was the documentation enough for you to sort this out?

**Reply by NeoCHoZen** [reply]
Thank you Daniel for your help with this. I'm quickly learning and understanding what's going on here. Even though I'm a little overwhelmed by the sheer amount of information out there, I can see what you are doing here and I want to use your stuff for a good while and see what I can deem from it. I'm a software QA by trade so I can also see this a training and testing session for myself and report bugs they come. I saw on a post on another forum where you imply you have the ability to supply the original BBSHD firmware if needed. I would like to personally acquire this since I failed to create a backup of my original firmware prior to flashing. Yeah I know stupid. So can I try that please? If you can. I have the 1.5 BBSHD with a 48v battery. Yeah I had a post on the FB group (Tim Delongchamp) where you replied to me on some stuff but for some reason I can't find it. So regardless, is this something you can supply me with? If anything it would be to just have it if I wish to level set back to stock and start fresh. This would be firmware of the 1.5 version. I'm curious and a quick question while you ponder this post....resellers don't have any control over firmware right? Or if they do, what's the likely hood of an Amazon reseller doing anything like that? Would it be a good idea to upgrade my controller? Luna or others? I'm open to suggestions....that guy that has that new controller he's pushing (can't find or remember the specifics at the moment).Thank you for your time responding to my post. I hope we can work together but understand I'm learning but I'm a quick learner...especially when I make mistakes :D 🤙🤙🤙🤙

**Reply by danielnilsson9** [reply]
Hi again!

Checkout this thread on endless sphere:
https://endless-sphere.com/forums/viewtopic.php?f=28&t=100064

This seems to be what most are using:
https://endless-sphere.com/forums/viewtopic.php?f=28&t=100064&start=100#p1591048

There are multiple versions of the original firmware posted for both 48V and 52V versions. As far as I know resellers can be supplied with firmware updates by Bafang but most probably never touches the original firmware shipped on the controller.

Yes you can update your controller, no issue, you will always be able to go back to an original firmware. The upcoming 1.1 versions contains quite some fixes and is more polished, so if you are not in a hurry I would probably had waited for that. Should be released next week if no big issues are reported with the BBS02 implementation.

**Reply by danielnilsson9** [reply]
1.1 has now been released!

**Comment by Delphis1982** [comment]
Hi,

I'm new here and I don't know, how this platform works, so I have to apologize, if I make some mistakes with my questions.

Is there any possibility, to compile the Config Tool for 32 Bit Windows?
I'm using an Acturion Durios T7M with Windows 7 x32.
I use it for mobile programming on the road.
Usually I prefer Windows XP (I have the Tablet PC Edition) but since this tool is written in NET Desktop Runtime
I have no chance to run XP anymore. The classic Config Tool from Bafang even run on Windows 95 (tried), but it's not compatible of course.
I never compiled some software from this platform.
Is there a chance to get a precompiled binary for x32 or some instructions with needed tools to compile myself?

Regards
David

**Reply by danielnilsson9** [reply]
Try this x86 build, no idea if it works on Windows 7 though, some other guy had trouble with that.

You can edit bbs-fw-tool.csproj, change RuntimeIdentifier->win-x64 to win-x86 and then to a publish from Visual Studio and it will produce a x86 binary.

[BBSFWTool.zip](https://github.com/danielnilsson9/bbs-fw/files/9718151/BBSFWTool.zip)

**Reply by nforakis** [reply]
Looks like this is flagged as a virus.

**Reply by danielnilsson9** [reply]
Weird, do you mean by chrome? It probably flagged it since it has not been downloaded many times, seems to do that often. If you are unsure, please build it yourself according to the instructions above, you will need to download Visual Studio.

**Reply by nforakis** [reply]
No, windows AV.

**Reply by nforakis** [reply]
Detected: Trojan:Script/Wacatac.B!ml

**Reply by danielnilsson9** [reply]
Ok, I cannot explain, not flagged if i scan it on my machine, Windows 10.

**Reply by Zensaition** [reply]
@noforakis Id either reinstall windows or just upgrade to windows 11 on a different machine...

**Comment by Delphis1982** [comment]
I received the file and it's clean.
At least it works on W7. Tomorrow I will try it on XP.
Thank you.
I think the problem is, that the file is completely unknown by the
signatures because of self compiling.


Daniel Nilsson ***@***.***> schrieb am Mi., 5. Okt. 2022,
20:56:
